### PR TITLE
cubical subtype

### PR DESCRIPTION
**Pull request by lunalunaa**
## Tasks: implement cubical subtype
- [x] syntax definitions for `Sub`, `inS`, `outS`
- [ ] typing rules
- [ ] computation rules
- [ ] rewrite tests and reimplement `coe` due to subtypes

Will update progress here.

**Note: this branch includes preliminary code for `hcomp`. This will be cleaned up because we are using cubical subtype now.**

### CODE REVIEW

**Comment by ice1000** [comment]
How to implement coe witu subtype

**Comment by ice1000** [comment]
Did you mean in the return type?

**Comment by lunalunaa** [comment]
> Did you mean in the return type?

Hey, can we have this type of extended discussions on Space instead? I don't want to receive email notifications for minor replies. Also, I prefer to be messaged on Eastern Time 9am to 7pm Monday to Friday. Sorry about that.

**Comment by imkiva** [comment]
It is hard to follow your preference since "Eastern Time 9am to 7pm" means Beijing Time 20:00 to 30:00. You might want to adjust your notification settings and it is totally OK if you choose to ignore messages when you are off work. Please, not all developers are in the same time zone as you are and both GitHub and Space have the option to turn off certain notifications. Sorry about that.

**Comment by ice1000** [comment]
image

User -> Settings

There will be options to have GitHub not to send emails

**Comment by ice1000** [comment]
Closing in favor of #826 

Note: the code for hcomp is not yet ported.
